packages/calcite-components/src/components/table/table.tsx

```diff
@@ -0,0 +1,524 @@
+
+  /** Specifies the page size of the component. When `true`, renders `calcite-pagination` */
+  @Prop({ reflect: true }) pageSize = 0;
+
+  /** Specifies the size of the component. */
+  @Prop({ reflect: true }) scale: Scale = "m";
+
+  /** Specifies the selection mode of the component. */
+  @Prop({ reflect: true }) selectionMode: Extract<"none" | "multiple" | "single", SelectionMode> =
+    "none";
+
+  /** When `true`, displays zebra styling in the component. */
+  @Prop({ reflect: true }) zebra = false;
+
+  @Watch("groupSeparator")
+  @Watch("numbered")
+  @Watch("numberingSystem")
+  @Watch("pageSize")
+  @Watch("scale")
+  @Watch("selectionMode")
+  handleNumberedChange(): void {
+    this.updateRows();
+  }
+
+  /**
+   * Specifies the component's selected items.
+   *
+   * @readonly
+   */
+  @Prop({ mutable: true }) selectedItems: HTMLCalciteTableRowElement[] = [];
+
+  /**
+   * Made into a prop for testing purposes only
+   *
+   * @internal
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messages: TableMessages;
+
+  /**
+   * Use this property to override individual strings used by the component.
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messageOverrides: Partial<TableMessages>;
+
+  @Watch("messageOverrides")
+  onMessagesChange(): void {
+    /* wired up by t9n util */
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteTableElement;
+
+  @State() colCount = 0;
+
+  @State() pageStartRow = 1;
+
+  @State() selectedCount = 0;
+
+  /* Workaround for Safari https://bugs.webkit.org/show_bug.cgi?id=258430 https://bugs.webkit.org/show_bug.cgi?id=239478 */
+  @State() readCellContentsToAT: boolean;
+
+  @State() defaultMessages: TableMessages;
+
+  @State() effectiveLocale = "";
+
+  @Watch("effectiveLocale")
+  effectiveLocaleChange(): void {
+    updateMessages(this, this.effectiveLocale);
+  }
+
+  private paginationEl: HTMLCalcitePaginationElement;
+
+  private allRows: HTMLCalciteTableRowElement[];
+
+  private headRows: HTMLCalciteTableRowElement[];
+
+  private bodyRows: HTMLCalciteTableRowElement[];
+
+  private footRows: HTMLCalciteTableRowElement[];
+
+  private tableHeadSlotEl: HTMLSlotElement;
+
+  private tableBodySlotEl: HTMLSlotElement;
+
+  private tableFootSlotEl: HTMLSlotElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  async componentWillLoad(): Promise<void> {
+    setUpLoadableComponent(this);
+    await setUpMessages(this);
+    this.readCellContentsToAT = /safari/i.test(getUserAgentString());
```

Comment:
@Can you add the following comment from [`interactive.ts`](https://github.com/Esri/calcite-design-system/blob/271d9855eef4aa94cb7131381c98ab03eea57e4e/packages/calcite-components/src/utils/interactive.ts#L29) here?

```ts
// ⚠️ browser-sniffing is not a best practice and should be avoided ⚠️
```